src/ConversationList/ConversationList.vala

```diff
@@ -203,9 +217,15 @@ public class Mail.ConversationList : Gtk.Box {
             }
         });
 
-        hide_read_switch.toggled.connect (() => load_folder.begin (folder_info_per_account));
+        hide_read_switch.notify["active"].connect (() => reload_active_folder ());
 
-        hide_unstarred_switch.toggled.connect (() => load_folder.begin (folder_info_per_account));
+        hide_unstarred_switch.notify["active"].connect (() => reload_active_folder ());
```

Comment:
I don't love to see refactoring changes also mixed in with feature changes unless its absolutely necessary. Is there a reason you changed this from connecting to `toggled` to notifying on the `active` property?

Reply:
I think I originally switched from `toggled` to `notify["active"]` while experimenting with manual persistence, before switching to `Settings.bind`. But this may not be needed now, effectively.